### PR TITLE
LPS-196697 Show error of File too big to preview for videos and audios

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
How to test it

1. Enable previews for Video/Audio
2. Go to : System Settings > D&M > File Entries >  Previewable Processor Maximum Size)
3. change the value and update
4. Upload a video/audio with a side bigger than the value introduced

image

- LPS-196697 check for preview generation exceptions on video preview
- LPS-196697 if video is not supported and has no video preview, fail fast
- LPS-196697 show message of file big for audio checking the support of the mimeType for the generation of the audio.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

File: modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/preview/DLVideoDLPreviewRendererProvider.java [hunk 1]
```diff
@@ -63,6 +78,28 @@ public DLPreviewRenderer getThumbnailDLPreviewRenderer(
...
+		}
+
+		if (!_videoProcessor.hasVideo(fileVersion)) {
+			if (!DLProcessorRegistryUtil.isPreviewableSize(fileVersion)) {
+				throw new DLPreviewSizeException();
+			}
+
+			throw new DLPreviewGenerationInProcessException();
+		}
+	}
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/preview/DLVideoDLPreviewRendererProvider.java:98]: Can you name this method `_validateFileVersion`? In any case, validation methods should start with `_validate*`

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/preview/DLVideoDLPreviewRendererProvider.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/preview/DLVideoDLPreviewRendererProvider.java:98]: the name is the same on the audio (old code), thats why this name.

I will rename both

Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/preview/DLVideoDLPreviewRendererProvider.java:98]: Hi @adolfopa 

The name _checkForPreviewGenerationExceptions is used on all the _RendererProvider_ so I will not change it unless you want me to change it on all of them.

Comment (adolfopa) [reply modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/preview/DLVideoDLPreviewRendererProvider.java:98]: If there's precedent, then it is fine. Keep it as is.

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by austinchiang** [comment]
Checking failures

**Comment by austinchiang** [comment]
@AliciaGarciaGarcia  A file preview test is failing and looks related https://testray.liferay.com/home/-/testray/case_results/2428324882. Part of the issue is that the FF was not applied to the test. I will try to fix what I can, repush this and run ci:test:document

**Comment by austinchiang** [comment]
ci:test:document

**Comment by AliciaGarciaGarcia** [comment]
> @AliciaGarciaGarcia A video preview test is failing and looks related https://testray.liferay.com/home/-/testray/case_results/2428324882. Part of the issue is that the FF was not applied to the test. I will try to fix what I can, repush this and run ci:test:document

But the copy feature (the FF) should not be related with this 😕 🤔 
Maybe the exception (if on test are catching it) or the message changes, but the need of the FF make no sense to me

I will check it too.


Thanks!

**Comment by austinchiang** [comment]
> > @AliciaGarciaGarcia A video preview test is failing and looks related https://testray.liferay.com/home/-/testray/case_results/2428324882. Part of the issue is that the FF was not applied to the test. I will try to fix what I can, repush this and run ci:test:document
> 
> But the copy feature (the FF) should not be related with this 😕 🤔 Maybe the exception (if on test are catching it) or the message changes, but the need of the FF make no sense to me
> 
> I will check it too.
> 
> Thanks!

The copy feature/FF seems to be in a different pull. This one is related to video previews and the FF is under https://liferay.atlassian.net/browse/LPS-175868 unless I am mistaken. Anyway I will check it

**Comment by austinchiang** [comment]
@AliciaGarciaGarcia external video shortcut previews don't seem to be working properly in this pull. I checked with Facebook/Youtube/Twitch and while the preview appears in the form field when you're first adding the shortcut the preview does not render after you publish and view the content (see attached video). I thought it might have something to do with the Previewable Processor Maximum Size, but after increasing that limit 1000x the preview still does not appear.

I also checked against master and the video previews are working fine.


[Screencast from 09-22-2023 01:05:48 PM.webm](https://github.com/liferay-lima/liferay-portal/assets/8635473/df2b8baa-304a-4f7e-ad8d-c17db8fdb892)

**Comment by AliciaGarciaGarcia** [comment]
Thanks @austinchiang 

I did not take into account the shortcut!

I will add a fix

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:document

**Comment by Jeremy-Ch** [comment]
Hi @AliciaGarciaGarcia Please help to resolve the conflicts. Thanks.

**Comment by AliciaGarciaGarcia** [comment]
rebased

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:document

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

**Comment by Jeremy-Ch** [comment]
The failures are unrelated

**Comment by Jeremy-Ch** [comment]
ci:forward:force

**Comment by Jeremy-Ch** [comment]
ci:test:stable

**Comment by Jeremy-Ch** [comment]
ci:forward:force

**Pull request closed without merging**